tests/core/mempool/test_mempool_manager.py

```diff
@@ -329,3 +330,51 @@ async def test_sb_twice_with_eligible_coin_and_different_spends_order() -> None:
     assert result == (expected_cost, MempoolInclusionStatus.PENDING, Err.MEMPOOL_CONFLICT)
     assert mempool_manager.get_spendbundle(sb_name) == sb
     assert mempool_manager.get_spendbundle(reordered_sb_name) is None
+
+
+@pytest.mark.asyncio
+async def test_get_items_not_in_filter() -> None:
+    mempool_manager = await instantiate_mempool_manager(get_coin_record_for_test_coins)
+    conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1]]
+    _, sb1_name, _ = await generate_and_add_spendbundle(mempool_manager, conditions)
+    mempool_item1 = mempool_manager.get_mempool_item(sb1_name)
+    conditions2 = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 2]]
+    _, sb2_name, _ = await generate_and_add_spendbundle(mempool_manager, conditions2, TEST_COIN2)
+    mempool_item2 = mempool_manager.get_mempool_item(sb2_name)
+    conditions3 = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 3]]
+    _, sb3_name, _ = await generate_and_add_spendbundle(mempool_manager, conditions3, TEST_COIN3)
+    mempool_item3 = mempool_manager.get_mempool_item(sb3_name)
+
+    # Don't filter anything
+    empty_filter = PyBIP158([])
+    result = await mempool_manager.get_items_not_in_filter(empty_filter)
+    assert result == [mempool_item3, mempool_item2, mempool_item1]
+
+    # Filter everything
+    full_filter = PyBIP158([bytearray(sb1_name), bytearray(sb2_name), bytearray(sb3_name)])
+    result = await mempool_manager.get_items_not_in_filter(full_filter)
+    assert result == []
+
+    # Negative limit
+    result = await mempool_manager.get_items_not_in_filter(empty_filter, limit=-1)
+    assert result == [mempool_item3, mempool_item2, mempool_item1]
```

Comment:
this return value seems a bit odd. is specifying a negative limit sensible? It would seem to make more sense to throw an exception. Maybe `assert` that `limit >= 0`

Reply:
That's exactly why I left that path covered here. I didn't know whether this behavior was intended or not so I thought to expose it.

Reply:
you might be able to find out whether it's deliberate or not by reviewing git history and call sites. I suspect it's not, and that we should turn it into an error